File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-0.cpp
class Solution {
public:
    unordered_map<char, vector<char>> hm = {
        {'2', {'a', 'b', 'c'}},
        {'3', {'d', 'e', 'f'}},
        {'4', {'g', 'h', 'i'}},
        {'5', {'j', 'k', 'l'}},
        {'6', {'m', 'n', 'o'}},
        {'7', {'p', 'q', 'r', 's'}},
        {'8', {'t', 'u', 'v'}},
        {'9', {'w', 'x', 'y', 'z'}},
    };
    string digits;
    vector<string> res;

    vector<string> letterCombinations(string digits) {
        if(digits == "") {
            return res;
        }
        this->digits = digits;
        string state = "";

        foo(0, state);
        return res;
    }

    void foo(int idx, string& state) {
        if(idx == digits.length()) {
            res.push_back(state);
            return;
        }

        auto arr = hm[digits[idx]];
        for(int j = 0; j < arr.size(); j++) {
            state += arr[j];
            foo(idx+1, state);
            state.pop_back();
        }
    }
};

File: Data Structures & Algorithms/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/submission-0.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        res = 0
        sum = 0
        i = 0
        while i < k:
            sum += arr[i]
            if i == k-1 and sum / k >= threshold:
                res += 1
            i += 1
        print(i)
        
        while i < len(arr):
            sum += arr[i]
            sum -= arr[i - k]
            if sum / k >= threshold:
                res += 1
            i += 1
        return res